src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Component.targets

```diff
@@ -0,0 +1,285 @@
+         MainEntryPoint="$(StartupObject)"
+         ModuleAssemblyName="$(ModuleAssemblyName)"
+         NoConfig="true"
+         NoLogo="$(NoLogo)"
+         NoStandardLib="$(NoCompilerStandardLib)"
+         NoWin32Manifest="$(NoWin32Manifest)"
+         Optimize="$(Optimize)"
+         Deterministic="$(Deterministic)"
+         PublicSign="$(PublicSign)"
+         OutputAssembly="@(_RazorComponentDeclarationAssembly)"
+         Platform="$(PlatformTarget)"
+         Prefer32Bit="$(Prefer32Bit)"
+         PreferredUILang="$(PreferredUILang)"
+         ProvideCommandLineArgs="$(ProvideCommandLineArgs)"
+         References="@(ReferencePathWithRefAssemblies)"
+         ReportAnalyzer="$(ReportAnalyzer)"
+         Resources="@(CompiledLicenseFile)"
+         ResponseFiles="$(CompilerResponseFile)"
+         RuntimeMetadataVersion="$(RuntimeMetadataVersion)"
+         SharedCompilationId="$(SharedCompilationId)"
+         SkipCompilerExecution="$(SkipCompilerExecution)"
+         Sources="@(_RazorComponentDeclaration);@(Compile)"
+         SubsystemVersion="$(SubsystemVersion)"
+         TargetType="$(OutputType)"
+         ToolExe="$(CscToolExe)"
+         ToolPath="$(CscToolPath)"
+         TreatWarningsAsErrors="$(TreatWarningsAsErrors)"
+         UseHostCompilerIfAvailable="$(UseHostCompilerIfAvailable)"
+         UseSharedCompilation="$(UseSharedCompilation)"
+         Utf8Output="$(Utf8Output)"
+         VsSessionGuid="$(VsSessionGuid)"
+         WarningLevel="$(WarningLevel)"
+         WarningsAsErrors="$(WarningsAsErrors)"
+         WarningsNotAsErrors="$(WarningsNotAsErrors)"
+         PathMap="$(PathMap)"
+         SourceLink="$(SourceLink)">
+      <Output TaskParameter="CommandLineArgs" ItemName="CscCommandLineArgs" />
+    </Csc>
+
+    <ItemGroup>
+      <FileWrites Include="@(_RazorComponentDeclarationAssembly)" Condition="Exists('@(_RazorComponentDeclarationAssembly)')" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_ResolveComponentRazorGenerateInputs">
+    <ItemGroup>
+      <!-- Add RazorComponent files to the set of files to be code-gened. -->
+      <RazorGenerateWithTargetPath Include="@(RazorComponentWithTargetPath)" />
+    </ItemGroup>
+
+    <!-- Point to the declaration assembly to perform taghelper discovery -->
+    <PropertyGroup>
+      <_RazorGenerateTargetReferenceAssembly>@(_RazorComponentDeclarationAssembly->Metadata('FullPath'))</_RazorGenerateTargetReferenceAssembly>
+    </PropertyGroup>
+  </Target>
+
+  <!-- 
+    Once RazorCompileComponentDeclaration is completed, we have all the inputs required for code generation of
+    RazorComponents and RazorGenerate items. We'll initialize the inputs for RazorGenerate and invoke it.
+    This should result in RazorGenerate \ RazorTagHelper targets no-oping when it get invoked as 
+    part of PrepareForRun \ PrepareForPublish for non-components (.cshtml) files.
+   -->
+  <Target
+    Name="RazorGenerateComponentDefinition"
```

Comment:
Here's the interesting bit - the Sdk calls RazorGenerate after setting up two pieces of state:

1) Adds `RazorComponents` to `RazorGenerateWithTargetPath` so they generate together. I could combine the list in the `RazorGenerate` targets instead to prevent dirtying the collection

2) Points the RazorTagHeper target to use the declaration assembly that we compiled as source.

In a non-component case, this target does not get called so everything carries on as normal. In component case, msbuild takes care of not re-invoking the targets after build